File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/ConformingContainerAntipattern/_3_ConformingContainer/Services/MsSqlBasedRepository.cs
using ConformingContainerAntipattern._3_ConformingContainer.Interfaces;

namespace ConformingContainerAntipattern._3_ConformingContainer.Services
{
  public interface IRepository
  {
    void Save(AcmeMessage message);
  }

  public class MsSqlBasedRepository : IRepository
  {
    private readonly DataDestination _sqlDataDestination = ApplicationRoot.Context.Resolve<SqlDataDestination>();

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}

File: Java/dependency-injection/service-locator/src/main/java/ServiceLocatorAntipattern/Services/MsSqlBasedRepository.java
package ServiceLocatorAntipattern.Services;

import ServiceLocatorAntipattern.ApplicationRoot;
import ServiceLocatorAntipattern.Interfaces.AcmeMessage;
import ServiceLocatorAntipattern.Interfaces.DataDestination;

public class MsSqlBasedRepository implements IRepository {
  private final DataDestination _sqlDataDestination = ApplicationRoot.context.getComponent(SqlDataDestination.class);

  public void save(AcmeMessage message) {
    message.writeTo(_sqlDataDestination);
  }
}
